src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs

```diff
@@ -21,6 +21,17 @@ public static void OnExecute(this CommandLineApplication app, Action action)
                     return 0;
                 });
 
+        public static CommandOption Option(this CommandLineApplication command, string template, string description)
+            => command.Option(
+                template,
+                description,
+                template.IndexOf('<') != -1
+                    ? template.EndsWith(">...") ? CommandOptionType.MultipleValue : CommandOptionType.SingleValue
```

Comment:
This looks odd, could you explain?

Reply:
This is a copy / paste from https://github.com/aspnet/AspNetCore/pull/13323/files#diff-e22cb33793ce6707b3111cac914249dfL8 (the old `CommandLineApplicationExtensions` used in these tools). The idea is to infer the `CommandOptionType` from the template. It's a compact version of the template parsing `CommandLineApplication` does when trying to determine the name of an option's value. The three cases are
- no '<' indicates the option has no value
- if there is a '<', ">..." indicates the option accepts multiple values
- otherwise, the option accepts a single value

Yes, the code ignores the possibility of '<' without a matching '>' but the `CommandLineApplication` template will throw in that case IIRC. If not, it's a general bug in the Extensions repo.